dot/network/messages/warp_sync.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package messages
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// WarpProofRequest is a struct for p2p warp proof request
+type WarpProofRequest struct {
+	Begin common.Hash
+}
+
+// Decode decodes the message into a WarpProofRequest
+func (wsr *WarpProofRequest) Decode(in []byte) error {
```

Comment:
```suggestion
func (wpr *WarpProofRequest) Decode(in []byte) error {
```
same for below receivers